opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -525,7 +598,12 @@ def tracer() -> Tracer:
 
     if _TRACER is None:
         # pylint:disable=protected-access
-        _TRACER = loader._load_impl(Tracer, _TRACER_FACTORY)
+        try:
+            _TRACER = loader._load_impl(Tracer, _TRACER_FACTORY)  # type: ignore
```

Comment:
```suggestion
            _TRACER = loader._load_impl(DefaultTracer, _TRACER_FACTORY)  # type: ignore
```

Would this do the same thing as the try/catch?

Reply:
Using a `DefaultTracer` would mean any implementation would be extending `DefaultTracer` rather than the `Tracer` interface, which means we would lose the value of using abstract methods to signal what methods are expected to be implemented.

Reply:
I think this `try:` catch should really be moved to the loader. Probably some reflection will be required (or a class attribute `DEFAULT_IMPLEMENTATION_CLASS` linking the two (alternatively `INTERFACE_CLASS` in the other direction)

Reply:
But if I were a vendor, I would definitely inherit from DefaultTracer instead of Tracer because I don't want my code to crash the customer's app if they start calling a new method that my Tracer doesn't (yet) implement. Which was also one argument by me (and I think @carlosalberto agreed) against this ABC/Default split in #66.

Reply:
I think that's a really good argument that almost convinces me against ABCs. I'm +1 to doing that work in this PR, or going back to switching to no-ops once we have all of this standardized.

Reply:
Yeah, definitely same feeling about vendors ending up inheriting from `DefaultTracer`.